sources/code/main/modules/extensions.ts

```diff
@@ -31,32 +38,30 @@ async function parseImports(cssString: string, importCalls:string[], maxTries=5)
       promises.push(Promise.reject(new Error("Circular reference in CSS imports are disallowed!")));
       return;
     }
-    importCalls.push(file);
-    promises.push(fetchOrRead(file)
+    promises.push(fetchOrRead(file, prevImport)
       .then(data => {
         if (data.download)
           return data.download.then(data => data.text());
         else
           return data.read.then(data => data.toString());
       })
+      .then(content => parseImports(content, [...importCalls, file], maxTries, dirname(file)))
```

Comment:
I don't like calling the recursion for an each import found in a single file, given we loop over them anyway. I feel this brings more redundant calls to the `parseImports`, and therefore harms a bit the performance and the possibly makes the application to also use a bit more of the memory than it is actually needed. Even tho there's a condition that makes the function to stop right after there's no more imports to resolve, a stack frame function takes is still likely allocated and some definitions and work is still done before that happens. Moreover, the previous call to `parseImports` keep the recursion a bit more linear, and that's a good thing if we want to analyse the case of a race condition to happen. So, I'm a bit unsure of this code for now. I guess I might take some steps to analyse it further.

Again, I would also eliminate `dirname(file)` there and more path resolving to a single place, than to split it all over the code and introduce more of the variables and/or function arguments that it is necessarily to have. This is all redundancy.

Reply:
Done